web/src/status.tsx

```diff
@@ -52,16 +52,32 @@ function runtimeStatus(r: UIResource): ResourceStatus {
 // 1) If there's a current or pending build, this is "pending".
 // 2) Otherwise, if there's a build error or runtime error, this is "error".
 // 3) Otherwise, we fallback to runtime status.
-function combinedStatus(res: UIResource): ResourceStatus {
-  let bs = buildStatus(res)
-  if (bs !== ResourceStatus.Healthy && bs !== ResourceStatus.None) {
-    return bs
+function summaryStatus(res: UIResource): ResourceStatus {
+  return summaryStatusFromAllStatuses({
+    buildStatus: buildStatus(res),
+    runtimeStatus: runtimeStatus(res),
+  })
+}
+
+function summaryStatusFromAllStatuses({
```

Comment:
Very open to function naming suggestions

Reply:
What do you think of keeping the original function but changing the signature so that you can pass it SidebarItem's info? e.g.:
```
function combinedStatus(buildStatus: ResourceStatus, runtimeStatus: ResourceStatus)
```
and then the two calls become
```
  resources.forEach((r) => {
    const status = combinedStatus(buildStatus(r), runtimeStatus(r))
```
and
```
  const statuses: ResourceStatus[] = props.items.map((item) =>
    combinedStatus(item.buildStatus, item.runtimeStatus)
  )
```

Reply:
i like this much better. thanks for the suggestion!